weldx/utility.py

```diff
@@ -94,6 +95,17 @@ def to_list(var) -> list:
     return [var]
 
 
+def to_pandas_time_index(time):
+    if not isinstance(time, np.ndarray):
+        if not isinstance(time, list):
+            time = [time]
+        time = np.array(time)
+
+    if np.issubdtype(time.dtype, np.datetime64):
+        return pd.DatetimeIndex(time)
+    return pd.TimedeltaIndex(time)
+
+
```

Comment:
this should also be a good place to convert `pint.Quantities` 

```python
def to_pandas_time_index(time):
    if isinstance(time, pint.Quantity):
        pint_unit = "s" # using low base unit could cause rounding errors
        try:
            return pd.TimedeltaIndex(data=time.to(pint_unit).magnitude,unit=pint_unit)
        except TypeError:
            return pd.TimedeltaIndex(data=[time.to(pint_unit).magnitude],unit=pint_unit)
        except Exception as e:
            raise e

    if not isinstance(time, np.ndarray):
        if not isinstance(time, list):
            time = [time]
        time = np.array(time)
    if np.issubdtype(time.dtype, np.datetime64):
        return pd.DatetimeIndex(time)
    return pd.TimedeltaIndex(time)
```

Reply:
Done